File: client/makc2021-react-redux-ts/src/Layer5/Pages/DummyMain/Item/Queries/Get/DummyMainItemPageGetQueryInput.ts
// Copyright (c) 2021 Maxim Kuzmin. All rights reserved. Licensed under the MIT License.

import {
  createDomainItemGetQueryInput,
  DomainItemGetQueryInput
} from 'src/Layer4/Domains/DummyMain/Queries/Item/Get/DomainItemGetQueryInput';

/**
 * Входные данные запроса на получение страницы сущности "DummyMain".
 */
export interface DummyMainItemPageGetQueryInput {
  /**
   * Элемент.
   */
  item: DomainItemGetQueryInput;
}

/**
 * Создать входные данные запроса на получение страницы сущности "DummyMain".
 * @returns Входные данные запроса на получение страницы сущности "DummyMain".
 */
export function createDummyMainItemPageGetQueryInput(): DummyMainItemPageGetQueryInput {
  return {
    item: createDomainItemGetQueryInput()
  } as DummyMainItemPageGetQueryInput;
}

File: server/src/Makc2021.Layer5.Sql.Server/Pages/DummyMain/Item/Queries/Get/DummyMainItemPageGetQueryInput.cs
﻿// Copyright (c) 2021 Maxim Kuzmin. All rights reserved. Licensed under the MIT License.

using DummyMainDomainItemGetQueryInput = Makc2021.Layer4.Sql.Domains.DummyMain.Queries.Item.Get.DomainItemGetQueryInput;

namespace Makc2021.Layer5.Sql.Server.Pages.DummyMain.Item.Queries.Get
{
    /// <summary>
    /// Входные данные запроса на получение страницы сущности "DummyMain".
    /// </summary>
    public class DummyMainItemPageGetQueryInput
    {
        #region Properties

        /// <summary>
        /// Входные данные запроса на получение элемента в домене "DummyMain".
        /// </summary>
        public DummyMainDomainItemGetQueryInput InputOfDummyMainDomainItemGetQuery { get; } = new();

        #endregion Properties
    }
}
